### ISSUE TITLE
Make FinalFieldCouldBeStatic detect constant variable

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3679]
**Affects PMD Version:6.41.0** 

**Rule:FinalFieldCouldBeStatic**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#finalfieldcouldbestatic>

**Description:**
Hi, I found a false negative about the rule FinalFieldCouldBeStatic. Please refer to the following case, PMD should report a warning a line 3 because `b` is assigned by a compile-time constant, but it doesn't.

**Code Sample demonstrating the issue:**
```java
class Clazz {
  public static final int a = 10;
  public final int b = a;
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
